src/logic/relation.lean

```diff
@@ -9,6 +9,36 @@ import tactic.basic
 
 variables {α : Type*} {β : Type*} {γ : Type*} {δ : Type*}
 
+section ne_imp
+
+variable {r : α → α → Prop}
+
+lemma is_refl.reflexive [is_refl α r] : reflexive r :=
+λ x, is_refl.refl x
+
+/-- To show a reflexive relation `r : α → α → Prop` holds over `x y : α`,
+it suffices to show it holds when `x ≠ y`. -/
+lemma reflexive.rel_of_ne_imp (h : reflexive r) {x y : α} (hr : x ≠ y → r x y) : r x y :=
+begin
+  by_cases hxy : x = y,
+  { exact hxy ▸ h x },
+  { exact hr hxy }
+end
+
+/-- If a reflexive relation `r : α → α → Prop` holds over `x y : α`,
+then whether or not `x ≠ y`. -/
+lemma reflexive.ne_imp_iff (h : reflexive r) {x y : α} :
+  (x ≠ y → r x y) ↔ r x y :=
+⟨h.rel_of_ne_imp, λ hr _, hr⟩
+
+/-- If a reflexive relation `r : α → α → Prop` holds over `x y : α`,
+then whether or not `x ≠ y`. Unlike `reflexive.ne_imp_iff`, this uses `[is_refl α r] -/
```

Comment:
The sentences in these docstrings look incomplete to me. "then whether or not X" seems like it should not be the end of the sentence.